### PR TITLE
fix(medusa): Account for multiple inventory items in get-inventory

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- Account for multiple inventory items when getting inventory (setting quantity as the largest fulfillable amount when getting inventory)
- Adjust quantity accordingly when setting channel availability when getting inventory. 

**Why**
- in the initial implementation we assumed a 1 to 1 relationship from variant to inventory item.

### CODE REVIEW

File: packages/medusa/src/api/routes/admin/variants/get-inventory.ts [hunk 1]
```diff
@@ -118,9 +123,18 @@ export default async (req, res) => {
...
+            variantInventoryItems.map(async (variantInventory) => {
+              return (
+                // eslint-disable-next-line max-len
+                (await salesChannelInventoryService.retrieveAvailableItemQuantity(
+                  channel.id,
+                  variantInventory.inventory_item_id
+                )) / variantInventory.required_quantity
+              )
+            })
+          ))
```
Comment (srindom) [inline packages/medusa/src/api/routes/admin/variants/get-inventory.ts:137]: thought: now that we have this in two places should we consider extracting it in a util - or maybe add it as a method in the ProductVariantInventoryService?

File: packages/medusa/src/services/product-variant-inventory.ts [hunk 2]
```diff
@@ -677,6 +662,52 @@ class ProductVariantInventoryService extends TransactionBaseService {
...
+   * @param channelId Sales channel id to fetch availability for
+   * @returns The available quantity of the variant from the inventoryItems
+   */
+  async getVariantQuantityFromVariantInventoryItems(
+    variantInventoryItems: ProductVariantInventoryItem[],
+    channelId: string
+  ): Promise<number> {
+    if (
+      variantInventoryItems.length &&
+      variantInventoryItems.some(
```
Comment (adrien2p) [inline packages/medusa/src/services/product-variant-inventory.ts:682]: suggestion: Move that to a const

File: packages/medusa/src/services/product-variant-inventory.ts [hunk 3]
```diff
@@ -677,6 +662,52 @@ class ProductVariantInventoryService extends TransactionBaseService {
...
+   * Get the quantity of a variant from a list of variantInventoryItems
+   * The inventory quantity of the variant should be equal to the inventory
+   * item with the smallest stock, adjusted for quantity required to fulfill
+   * the given variant.
+   *
+   * @param variantInventoryItems List of inventoryItems for a given variant, These must all be for the same variant
+   * @param channelId Sales channel id to fetch availability for
+   * @returns The available quantity of the variant from the inventoryItems
+   */
+  async getVariantQuantityFromVariantInventoryItems(
```
Comment (adrien2p) [inline packages/medusa/src/services/product-variant-inventory.ts:676]: 🙏

**Review by adrien2p — APPROVED** [review]
LGTM, some suggestions

**Comment by olivermrbl** [comment]
@pKorsholm Can I get you to resolve conflicts, so we can merge? 🙏

**Comment by srindom** [comment]
@pKorsholm - don't forget this one

**Pull request merged**